src/explore-education-statistics-common/src/modules/table-tool/components/TimePeriodDataTable.tsx

```diff
@@ -296,17 +296,16 @@ const TimePeriodDataTableInternal = forwardRef<HTMLElement, Props>(
 
     // Only need to use first row's columnFilters
     // as they are the same for every row.
-    const columnHeaders = filteredCartesian[0].reduce<Header[]>(
-      (acc, column) => {
-        const filters = optimizeFilters(column.columnFilters, [
-          ...tableHeadersConfig.columnGroups,
-          tableHeadersConfig.columns,
-        ]);
-
-        return addFilters(acc, filters);
-      },
-      [],
-    );
+    const columnHeaders = filteredCartesian.length
+      ? filteredCartesian[0].reduce<Header[]>((acc, column) => {
+          const filters = optimizeFilters(column.columnFilters, [
+            ...tableHeadersConfig.columnGroups,
+            tableHeadersConfig.columns,
+          ]);
+
+          return addFilters(acc, filters);
+        }, [])
+      : [];
```

Comment:
Would suggest writing this to avoid the extra ternary:

```ts
const columnHeaders = (filteredCartesian[0] ?? []).reduce(...);
```

Isn't this going to end up with no column headers and a broken table anyway?

Reply:
You get the page rendering without the table. 
![Capture](https://user-images.githubusercontent.com/81572860/151815847-ce1eb7ed-3d51-4330-b22a-dda65a7ff865.PNG)

Reply:
Hmmm, not sure it's a good idea to have a caption for a completely empty table. Would suggest that we should render an error message to say that there was a problem rendering the table. 

Interestingly, this is what should already have been happening as `TimePeriodDataTable` wraps `TimePeriodDataTableInternal` with an error boundary. Why was this throwing a page error?

Reply:
Good point, I've changed it to show a warning message instead.

I don't think error boundaries work with server rendered pages (this component is used in the admin as well where presumably the boundary would work).

Reply:
Ah yes, that really sucks on Next's part! An alternative then might be to wrap it all with a try/catch in `TimePeriodDataTableInternal`, although I'm not 100% sure if this will work or not.